src/data/getDataBy.ts

```diff
@@ -51,13 +51,32 @@ export function getDataBy<
 
 			// Keys can be arrays (see STATUSES), handle it
 			const newKeys = ensureArray(val[by])
-			newKeys.forEach(key => map.set(key, val))
+			newKeys.forEach(key => {
+				if (map.has(key)) {
+					map.get(key)?.push(val)
+				} else {
+					map.set(key, [val])
+				}
```

Comment:
while races are _technically_ unlikely because >js runtime, i generally preference the get-once method;

```ts
let values = map.get(key)
if (values == null) {
	values = []
	map.set(key, values)
}
values.push(val)
```